File: src/libqtestsupport_widgets.cpp
#include <QEventPoint>
#include <QPoint>
#define WORKAROUND_INNER_CLASS_DEFINITION_QTest__QTouchEventSequence
#define WORKAROUND_INNER_CLASS_DEFINITION_QTest__QTouchEventWidgetSequence
#include <QWidget>
#include <qtestsupport_widgets.h>
#include "libqtestsupport_widgets.h"
#include "libqtestsupport_widgets.hxx"

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_new(const QTest__QTouchEventWidgetSequence* param1) {
    return new VirtualQTestQTouchEventWidgetSequence(*param1);
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Press(QTest__QTouchEventWidgetSequence* self, int touchId, const QPoint* pt) {
    QTest::QTouchEventWidgetSequence& _ret = self->press(static_cast<int>(touchId), *pt);
    // Cast returned reference into pointer
    return &_ret;
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Move(QTest__QTouchEventWidgetSequence* self, int touchId, const QPoint* pt) {
    QTest::QTouchEventWidgetSequence& _ret = self->move(static_cast<int>(touchId), *pt);
    // Cast returned reference into pointer
    return &_ret;
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Release(QTest__QTouchEventWidgetSequence* self, int touchId, const QPoint* pt) {
    QTest::QTouchEventWidgetSequence& _ret = self->release(static_cast<int>(touchId), *pt);
    // Cast returned reference into pointer
    return &_ret;
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Press3(QTest__QTouchEventWidgetSequence* self, int touchId, const QPoint* pt, QWidget* widget) {
    QTest::QTouchEventWidgetSequence& _ret = self->press(static_cast<int>(touchId), *pt, widget);
    // Cast returned reference into pointer
    return &_ret;
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Move3(QTest__QTouchEventWidgetSequence* self, int touchId, const QPoint* pt, QWidget* widget) {
    QTest::QTouchEventWidgetSequence& _ret = self->move(static_cast<int>(touchId), *pt, widget);
    // Cast returned reference into pointer
    return &_ret;
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Release3(QTest__QTouchEventWidgetSequence* self, int touchId, const QPoint* pt, QWidget* widget) {
    QTest::QTouchEventWidgetSequence& _ret = self->release(static_cast<int>(touchId), *pt, widget);
    // Cast returned reference into pointer
    return &_ret;
}

// Derived class handler implementation
QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Stationary(QTest__QTouchEventWidgetSequence* self, int touchId) {
    auto* vqtestqtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
    if (vqtestqtoucheventwidgetsequence && vqtestqtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
        return new QTest::QTouchEventWidgetSequence(vqtestqtoucheventwidgetsequence->stationary(static_cast<int>(touchId)));
    } else {
        return new QTest::QTouchEventWidgetSequence(((VirtualQTestQTouchEventWidgetSequence*)self)->stationary(static_cast<int>(touchId)));
    }
}

// Base class handler implementation
QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_QBaseStationary(QTest__QTouchEventWidgetSequence* self, int touchId) {
    auto* vqtestqtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
    if (vqtestqtoucheventwidgetsequence && vqtestqtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
        vqtestqtoucheventwidgetsequence->setQTest__QTouchEventWidgetSequence_Stationary_IsBase(true);
        return new QTest::QTouchEventWidgetSequence(vqtestqtoucheventwidgetsequence->stationary(static_cast<int>(touchId)));
    } else {
        return new QTest::QTouchEventWidgetSequence(((VirtualQTestQTouchEventWidgetSequence*)self)->stationary(static_cast<int>(touchId)));
    }
}

// Auxiliary method to allow providing re-implementation
void QTest__QTouchEventWidgetSequence_OnStationary(QTest__QTouchEventWidgetSequence* self, intptr_t slot) {
    auto* vqtestqtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
    if (vqtestqtoucheventwidgetsequence && vqtestqtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
        vqtestqtoucheventwidgetsequence->setQTest__QTouchEventWidgetSequence_Stationary_Callback(reinterpret_cast<VirtualQTestQTouchEventWidgetSequence::QTest__QTouchEventWidgetSequence_Stationary_Callback>(slot));
    }
}

// Derived class handler implementation
bool QTest__QTouchEventWidgetSequence_Commit(QTest__QTouchEventWidgetSequence* self, bool processEvents) {
    auto* vqtestqtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
    if (vqtestqtoucheventwidgetsequence && vqtestqtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
        return vqtestqtoucheventwidgetsequence->commit(processEvents);
    } else {
        return self->QTest::QTouchEventWidgetSequence::commit(processEvents);
    }
}

// Base class handler implementation
bool QTest__QTouchEventWidgetSequence_QBaseCommit(QTest__QTouchEventWidgetSequence* self, bool processEvents) {
    auto* vqtestqtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
    if (vqtestqtoucheventwidgetsequence && vqtestqtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
        vqtestqtoucheventwidgetsequence->setQTest__QTouchEventWidgetSequence_Commit_IsBase(true);
        return vqtestqtoucheventwidgetsequence->commit(processEvents);
    } else {
        return self->QTest::QTouchEventWidgetSequence::commit(processEvents);
    }
}

// Auxiliary method to allow providing re-implementation
void QTest__QTouchEventWidgetSequence_OnCommit(QTest__QTouchEventWidgetSequence* self, intptr_t slot) {
    auto* vqtestqtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
    if (vqtestqtoucheventwidgetsequence && vqtestqtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
        vqtestqtoucheventwidgetsequence->setQTest__QTouchEventWidgetSequence_Commit_Callback(reinterpret_cast<VirtualQTestQTouchEventWidgetSequence::QTest__QTouchEventWidgetSequence_Commit_Callback>(slot));
    }
}

// Derived class handler implementation
QEventPoint* QTest__QTouchEventWidgetSequence_Point(QTest__QTouchEventWidgetSequence* self, int touchId) {
    auto* vqtestqtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
    if (vqtestqtoucheventwidgetsequence && vqtestqtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
        return new QEventPoint(vqtestqtoucheventwidgetsequence->point(static_cast<int>(touchId)));
    }
    return {};
}

// Base class handler implementation
QEventPoint* QTest__QTouchEventWidgetSequence_QBasePoint(QTest__QTouchEventWidgetSequence* self, int touchId) {
    auto* vqtestqtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
    if (vqtestqtoucheventwidgetsequence && vqtestqtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
        vqtestqtoucheventwidgetsequence->setQTest__QTouchEventWidgetSequence_Point_IsBase(true);
        return new QEventPoint(vqtestqtoucheventwidgetsequence->point(static_cast<int>(touchId)));
    }
    return {};
}

// Auxiliary method to allow providing re-implementation
void QTest__QTouchEventWidgetSequence_OnPoint(QTest__QTouchEventWidgetSequence* self, intptr_t slot) {
    auto* vqtestqtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
    if (vqtestqtoucheventwidgetsequence && vqtestqtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
        vqtestqtoucheventwidgetsequence->setQTest__QTouchEventWidgetSequence_Point_Callback(reinterpret_cast<VirtualQTestQTouchEventWidgetSequence::QTest__QTouchEventWidgetSequence_Point_Callback>(slot));
    }
}

// Derived class handler implementation
QEventPoint* QTest__QTouchEventWidgetSequence_PointOrPreviousPoint(QTest__QTouchEventWidgetSequence* self, int touchId) {
    auto* vqtestqtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
    if (vqtestqtoucheventwidgetsequence && vqtestqtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
        return new QEventPoint(vqtestqtoucheventwidgetsequence->pointOrPreviousPoint(static_cast<int>(touchId)));
    }
    return {};
}

// Base class handler implementation
QEventPoint* QTest__QTouchEventWidgetSequence_QBasePointOrPreviousPoint(QTest__QTouchEventWidgetSequence* self, int touchId) {
    auto* vqtestqtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
    if (vqtestqtoucheventwidgetsequence && vqtestqtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
        vqtestqtoucheventwidgetsequence->setQTest__QTouchEventWidgetSequence_PointOrPreviousPoint_IsBase(true);
        return new QEventPoint(vqtestqtoucheventwidgetsequence->pointOrPreviousPoint(static_cast<int>(touchId)));
    }
    return {};
}

// Auxiliary method to allow providing re-implementation
void QTest__QTouchEventWidgetSequence_OnPointOrPreviousPoint(QTest__QTouchEventWidgetSequence* self, intptr_t slot) {
    auto* vqtestqtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
    if (vqtestqtoucheventwidgetsequence && vqtestqtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
        vqtestqtoucheventwidgetsequence->setQTest__QTouchEventWidgetSequence_PointOrPreviousPoint_Callback(reinterpret_cast<VirtualQTestQTouchEventWidgetSequence::QTest__QTouchEventWidgetSequence_PointOrPreviousPoint_Callback>(slot));
    }
}

void QTest__QTouchEventWidgetSequence_Delete(QTest__QTouchEventWidgetSequence* self) {
    delete self;
}

File: include/libqtestsupport_widgets.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html
pub const qtest__qtoucheventwidgetsequence = struct {
    /// New constructs a new QTest::QTouchEventWidgetSequence object.
    ///
    /// ``` param1: QtC.QTest__QTouchEventWidgetSequence ```
    pub fn New(param1: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_new(@ptrCast(param1));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html#press)
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32, pt: QtC.QPoint ```
    pub fn Press(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Press(@ptrCast(self), @intCast(touchId), @ptrCast(pt));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html#move)
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32, pt: QtC.QPoint ```
    pub fn Move(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Move(@ptrCast(self), @intCast(touchId), @ptrCast(pt));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html#release)
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32, pt: QtC.QPoint ```
    pub fn Release(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Release(@ptrCast(self), @intCast(touchId), @ptrCast(pt));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html#stationary)
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32 ```
    pub fn Stationary(self: ?*anyopaque, touchId: i32) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Stationary(@ptrCast(self), @intCast(touchId));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html#stationary)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, slot: fn (self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32) callconv(.c) QtC.QTest__QTouchEventWidgetSequence ```
    pub fn OnStationary(self: ?*anyopaque, slot: fn (?*anyopaque, i32) callconv(.c) QtC.QTest__QTouchEventWidgetSequence) void {
        qtc.QTest__QTouchEventWidgetSequence_OnStationary(@ptrCast(self), @as(isize, @bitCast(@intFromPtr(&slot))));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html#stationary)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32 ```
    pub fn QBaseStationary(self: ?*anyopaque, touchId: i32) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_QBaseStationary(@ptrCast(self), @intCast(touchId));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html#commit)
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, processEvents: bool ```
    pub fn Commit(self: ?*anyopaque, processEvents: bool) bool {
        return qtc.QTest__QTouchEventWidgetSequence_Commit(@ptrCast(self), processEvents);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html#commit)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, slot: fn (self: QtC.QTest__QTouchEventWidgetSequence, processEvents: bool) callconv(.c) bool ```
    pub fn OnCommit(self: ?*anyopaque, slot: fn (?*anyopaque, bool) callconv(.c) bool) void {
        qtc.QTest__QTouchEventWidgetSequence_OnCommit(@ptrCast(self), @as(isize, @bitCast(@intFromPtr(&slot))));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html#commit)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, processEvents: bool ```
    pub fn QBaseCommit(self: ?*anyopaque, processEvents: bool) bool {
        return qtc.QTest__QTouchEventWidgetSequence_QBaseCommit(@ptrCast(self), processEvents);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html#press)
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32, pt: QtC.QPoint, widget: QtC.QWidget ```
    pub fn Press3(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque, widget: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Press3(@ptrCast(self), @intCast(touchId), @ptrCast(pt), @ptrCast(widget));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html#move)
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32, pt: QtC.QPoint, widget: QtC.QWidget ```
    pub fn Move3(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque, widget: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Move3(@ptrCast(self), @intCast(touchId), @ptrCast(pt), @ptrCast(widget));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html#release)
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32, pt: QtC.QPoint, widget: QtC.QWidget ```
    pub fn Release3(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque, widget: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Release3(@ptrCast(self), @intCast(touchId), @ptrCast(pt), @ptrCast(widget));
    }

    /// Inherited from QTest::QTouchEventSequence
    ///
    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventsequence.html#point)
    ///
    /// Wrapper to allow calling virtual or protected method
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32 ```
    pub fn Point(self: ?*anyopaque, touchId: i32) QtC.QEventPoint {
        return qtc.QTest__QTouchEventWidgetSequence_Point(@ptrCast(self), @intCast(touchId));
    }

    /// Inherited from QTest::QTouchEventSequence
    ///
    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventsequence.html#point)
    ///
    /// Wrapper to allow calling base class virtual or protected method
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32 ```
    pub fn QBasePoint(self: ?*anyopaque, touchId: i32) QtC.QEventPoint {
        return qtc.QTest__QTouchEventWidgetSequence_QBasePoint(@ptrCast(self), @intCast(touchId));
    }

    /// Inherited from QTest::QTouchEventSequence
    ///
    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventsequence.html#point)
    ///
    /// Wrapper to allow overriding base class virtual or protected method
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, slot: fn (self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32) callconv(.c) QtC.QEventPoint ```
    pub fn OnPoint(self: ?*anyopaque, slot: fn (?*anyopaque, i32) callconv(.c) QtC.QEventPoint) void {
        qtc.QTest__QTouchEventWidgetSequence_OnPoint(@ptrCast(self), @as(isize, @bitCast(@intFromPtr(&slot))));
    }

    /// Inherited from QTest::QTouchEventSequence
    ///
    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventsequence.html#pointOrPreviousPoint)
    ///
    /// Wrapper to allow calling virtual or protected method
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32 ```
    pub fn PointOrPreviousPoint(self: ?*anyopaque, touchId: i32) QtC.QEventPoint {
        return qtc.QTest__QTouchEventWidgetSequence_PointOrPreviousPoint(@ptrCast(self), @intCast(touchId));
    }

    /// Inherited from QTest::QTouchEventSequence
    ///
    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventsequence.html#pointOrPreviousPoint)
    ///
    /// Wrapper to allow calling base class virtual or protected method
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32 ```
    pub fn QBasePointOrPreviousPoint(self: ?*anyopaque, touchId: i32) QtC.QEventPoint {
        return qtc.QTest__QTouchEventWidgetSequence_QBasePointOrPreviousPoint(@ptrCast(self), @intCast(touchId));
    }

    /// Inherited from QTest::QTouchEventSequence
    ///
    /// [Qt documentation](https://doc.qt.io/qt-6/qtest-qtoucheventsequence.html#pointOrPreviousPoint)
    ///
    /// Wrapper to allow overriding base class virtual or protected method
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence, slot: fn (self: QtC.QTest__QTouchEventWidgetSequence, touchId: i32) callconv(.c) QtC.QEventPoint ```
    pub fn OnPointOrPreviousPoint(self: ?*anyopaque, slot: fn (?*anyopaque, i32) callconv(.c) QtC.QEventPoint) void {
        qtc.QTest__QTouchEventWidgetSequence_OnPointOrPreviousPoint(@ptrCast(self), @as(isize, @bitCast(@intFromPtr(&slot))));
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QTest__QTouchEventWidgetSequence ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QTest__QTouchEventWidgetSequence_Delete(@ptrCast(self));
    }
};
